float32[] center_pix
float32 offset_
